### DISCUSSION TITLE
Network Issue anyone ??

### DISCUSSION THREAD

**Discussion by zibran4u** [PaddlePaddle/PaddleOCR#7024] [Q&A]
Am I the only one who is facing this network issue ??

Step.1) Installation:-
    python3 -m pip install paddlepaddle-gpu (Done)

Step.2) Wheel Package:-
    pip install "paddleocr>=2.0.1" # Recommend to use version 2.0.1+ (Done)

Step.3) paddleocr --image_dir image/license_plate.png --use_angle_cls True --lang en (Error)

requests.exceptions.ConnectionError: HTTPSConnectionPool(host='paddleocr.bj.bcebos.com', port=443): Max retries exceeded with url: /PP-OCRv3/english/en_PP-OCRv3_det_infer.tar (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fea92a5bef0>: Failed to establish a new connection: [Errno -2] Name or service not known',))

Also Tried :- 

:- https://paddleocr.bj.bcebos.com/PP-OCRv3/english/en_PP-OCRv3_rec_infer.tar (Site unreachable)

**Comment by littletomatodonkey** [comment]
Hi, it seems that your network can not link access the downlink link, you can download it mannualy and assign the model dir for your usage. 

The following attachment is the model files.


[rec_det_models.zip](https://github.com/PaddlePaddle/PaddleOCR/files/9230410/rec_det_models.zip)

**Reply by anidiatm41** [reply]
where to use this weight in the paddleocr repo?

**Reply by wqrao** [reply]
Hello, I am in trouble too, how to assign the model dir for my usage?

**Reply by kompaqt** [reply]
On windows the models are downloaded to `C:\Users\USERNAME\.paddlex\official_models`. So maybe just extract to this directory.

I copied models from one PC to another and it worked. It is probably in a different location on Linux.

**Comment by Evezerest** [comment]
Is there anyone else facing the same problem?

**Reply by anidiatm41** [reply]
yes, I am having same issue ..pls provide the steps to set the PPOCRLabel locally(offline)

**Reply by Evezerest** [reply]
As PPOCRLabel is built based on PP-OCR models, when you first run it, it will download the models automatically.  
We will update paddleocr whl package to fix this problem later, btw could you please tell us your location or country?  how many people may face the same problem?

**Reply by logan-markewich** [reply]
I'm having the same issue, located in Canada

**Reply by Kirandeephere** [reply]
Same problem, has it been fixed yet?

**Reply by navaneeth-aot** [reply]
configure your DNS server to Google public DNS
Google Primary DNS server: 8.8.8.8
Google Secondary DNS server: 8.8.4.4

**Comment by MustafaAlahmid** [comment]
I'm having the same problem 
any solution yet ?

**Comment by logan-markewich** [comment]
The models are being hosted on a server that seems to be unreachable to users in North America (I am unable to access in Canada). It would be great if this could be fixed

**Comment by np-n** [comment]
I stuck on same issue due to my location. It is automatically downloaded for Nepal but not for US. why?

**Reply by navaneeth-aot** [reply]
This is because DNS server Issue you should configure your DNS server to google public DNS
Google Primary DNS server: 8.8.8.8
Google Secondary DNS server: 8.8.4.4

**Comment by xldistance** [comment]
Find site-packages\paddleocr\ppocr\utils\network.py, and modify the following code
```
def download_with_progressbar(url, save_path):
    logger = get_logger()
    response = requests.get(url, stream=True,proxies={'https':'http://127.0.0.1:4378'})
```

**Reply by JemimaSherley** [reply]
Your solution was almost successful for me.
Using "**requests.get(url, verify=False)**" resolved the issue.

Here is the function that worked for me:
`def download_with_progressbar(url, save_path):` 
     `logger = get_logger()` 
     `response = requests.get(url, stream=True, verify=False)`

**Comment by akshat-kumar-alight** [comment]
For paddleocr =2.6.1.3, english, download the files from the det --> https://paddleocr.bj.bcebos.com/PP-OCRv3/english/en_PP-OCRv3_det_infer.tar

rec --> https://paddleocr.bj.bcebos.com/PP-OCRv3/english/en_PP-OCRv3_rec_infer.tar

cls --> https://paddleocr.bj.bcebos.com/dygraph_v2.0/ch/ch_ppocr_mobile_v2.0_cls_infer.tar

and place them in ".paddleocr" folder in "C:\Users<username><folder_structure>"the folder structure will be the location that you get in the error. For me it was following:

.paddleocr\whl\cls\ch_ppocr_mobile_v2.0_cls_infer\ch_ppocr_mobile_v2.0_cls_infer.tar
.paddleocr\whl\det\en\en_PP-OCRv3_det_infer\en_PP-OCRv3_det_infer.tar
.paddleocr\whl\rec\en\en_PP-OCRv3_rec_infer\en_PP-OCRv3_rec_infer.tar
and comment the line Lib\site-packages\paddleocr\ppocr\utils\network.py --> #download_with_progressbar(url, tmp_path)

This should help you run paddleocr.. Run the following lines to check

from paddleocr import PaddleOCR,draw_ocr

ocr = PaddleOCR(lang='en')

**Comment by navaneeth-aot** [comment]
I had the same issue. you need to reconfigure your DNS server to any public DNS server.
In my case, I have tried so many ways to solve the issue but always failed to find a proper solution. Later I found out that the issue was with my router setup. https://paddleocr.bj.bcebos.com/  I couldn't access this URL using my wifi network. 

As a solution for this, you need to reconfigure the router DNS server to any public DNS server. I have configured the Google DNS server and it works for me. 
Google Primary DNS server: 8.8.8.8
Google Secondary DNS server: 8.8.4.4

configure this DNS I hope this should do

**Comment by santosh9sanjeev** [comment]
not able to download the weights?

**Comment by nikitashiwach2001** [comment]
I am having the same problem. Any solution
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='paddleocr.bj.bcebos.com', port=443): Max retries exceeded with url: /PP-OCRv3/english/en_PP-OCRv3_det_infer.tar (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x70f643965ae0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
